configuration/src/main/java/io/camunda/configuration/beanoverrides/BrokerBasedPropertiesOverride.java

```diff
@@ -361,4 +369,58 @@ private void populateFromSasToken(final BrokerBasedProperties override) {
           .setSasToken(SasToken.fromSasTokenConfig(sasTokenConfig).toSasTokenConfig());
     }
   }
+
+  private void populateExporter(
+      final BrokerBasedProperties override,
+      final String exporterName,
+      final String className,
+      final String jarPath) {
+    final SecondaryStorage secondaryStorage =
+        unifiedConfiguration.getCamunda().getData().getSecondaryStorage();
+
+    final SecondaryStorageDatabase database;
+    if (SecondaryStorageType.elasticsearch == secondaryStorage.getType()) {
+      database =
+          unifiedConfiguration.getCamunda().getData().getSecondaryStorage().getElasticsearch();
+    } else if (SecondaryStorageType.opensearch == secondaryStorage.getType()) {
+      database = unifiedConfiguration.getCamunda().getData().getSecondaryStorage().getOpensearch();
+    } else {
+      // RDBMS and NONE are not supported.
+      return;
+    }
+
+    /* Load exporter config map */
+
+    final List<ExporterCfg> exporters =
+        override.getExporters().values().stream()
+            .filter(e -> e.getClassName().equals(className))
+            .toList();
+
+    final ExporterCfg exporter;
+    if (exporters.isEmpty()) {
+      exporter = new ExporterCfg();
+      exporter.setJarPath(jarPath);
+      exporter.setClassName(className);
+      exporter.setArgs(new LinkedHashMap<>());
+      override.getExporters().put(exporterName, exporter);
+    } else {
+      exporter = exporters.getFirst();
+    }
+
+    /* Override config map values */
+
+    final Map<String, Object> args = exporter.getArgs();
+    setArg(args, "connect.type", secondaryStorage.getType().name());
+    setArg(args, "connect.url", database.getUrl());
```

Comment:
:thinking: It's is sad that convertion from object to map did not work. It would be a pain to maintain if we use statically coded prefix. Let's keep it like this for now. But I would really like if you could figure out a way to populate it via configuration class. Otherwise, any rename and restructuring in the configuration class will break this wiring.